### PR TITLE
Added REMOVE PARTITIONING and ALTER TABLE t PARTITION BY

### PR DESCRIPTION
**Pull request by mjonss**
### What is changed, added or deleted? (Required)

Added `ALTER TABLE t REMOVE PARTITIONING` and `ALTER TABLE t PARTITION BY ...`

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [X] master (the latest development version)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):
https://github.com/pingcap/tidb/pull/42907
https://github.com/pingcap/tiflash/pull/7822
https://github.com/pingcap/tiflow/pull/9670

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: partitioned-table.md [hunk 1]
```diff
@@ -1025,6 +1025,16 @@ ALTER TABLE members OPTIMIZE PARTITION p0;
 ERROR 8200 (HY000): Unsupported optimize partition
 ```
 
+## How to add or remove partitioning from an existing table
```
Comment (qiancai) [inline partitioned-table.md:1028]:
```suggestion
## Convert a partitioned table to a non-partitioned table
```

File: partitioned-table.md [hunk 2]
```diff
@@ -1025,6 +1025,16 @@ ALTER TABLE members OPTIMIZE PARTITION p0;
...
+
+Remove partitioning from an already partitioned table:
+
+- `ALTER TABLE t REMOVE PARTITIONING` which will alter a partitioned table to a non-partitioned table by copying all rows and recreate the indexes on-line.
+
```
Comment (qiancai) [inline partitioned-table.md:1033]:
```suggestion
## Convert a non-partitioned table to a partitioned table
```

File: partitioned-table.md [hunk 3]
```diff
@@ -1025,6 +1025,16 @@ ALTER TABLE members OPTIMIZE PARTITION p0;
 ERROR 8200 (HY000): Unsupported optimize partition
 ```
```
Comment (qiancai) [inline partitioned-table.md:1027]: @mjonss please also add a section in this PR to describe how to modify one partition type to another. Thanks.

Comment (mjonss) [reply partitioned-table.md:1027]: I rephrased 'Convert a non-partitioned table to a partitioned table' into 'Partition an existing table' with some clarifying text.

File: partitioned-table.md [hunk 4]
```diff
@@ -1018,13 +1018,41 @@ Query OK, 0 rows affected (0.03 sec)
...
 
 ```sql
 ERROR 8200 (HY000): Unsupported optimize partition
 ```
 
+## Convert a partitioned table to a non-partitioned table
+
+Remove partitioning from an already partitioned table:
+
+- `ALTER TABLE members REMOVE PARTITIONING` which will alter a partitioned table to a non-partitioned table by copying all rows and recreate the indexes on-line.
```
Comment (qiancai) [inline partitioned-table.md:1032]:
````suggestion
To convert a partitioned table to a non-partitioned table, you can use the following statement, which removes the partitioning, copies all rows of the table, and recreates the indexes online for the table:

```sql
ALTER TABLE <table_name> REMOVE PARTITIONING
```

For example, to convert the `members` partitioned table to the non-partitioned table, you can execute the following statement:

```sql
ALTER TABLE members REMOVE PARTITIONING
```
````

File: partitioned-table.md [hunk 5]
```diff
@@ -1018,13 +1018,41 @@ Query OK, 0 rows affected (0.03 sec)
...
+
+## Partition an existing table
+
+Partition an existing table (regardless if it is already partitioned or not):
+
+- `ALTER TABLE t PARTITION BY ...` which will alter the table to a partitioned table by copying all rows and recreate the indexes on-line according to the partitioning scheme.
```
Comment (qiancai) [inline partitioned-table.md:1038]:
````suggestion
To partition an existing non-partitioned table or modify the partition type of an existing partitioned table, you can use the following statement, which copies all rows and recreates the indexes online according to the new partition definitions:

```sql
ALTER TABLE <table_name> PARTITION BY <new partition definitions>
```
````

File: partitioned-table.md [hunk 6]
```diff
@@ -1018,13 +1018,41 @@ Query OK, 0 rows affected (0.03 sec)
...
+
+Example:
```
Comment (qiancai) [inline partitioned-table.md:1040]:
```suggestion
Examples:

To convert the existing `members` table to a HASH partitioned table with 10 partitions, you can execute the following statement:
```

File: partitioned-table.md [hunk 7]
```diff
@@ -1018,13 +1018,41 @@ Query OK, 0 rows affected (0.03 sec)
...
+
+```sql
+ALTER TABLE members PARTITION BY HASH(id) PARTITIONS 10;
+```
+
+or
```
Comment (qiancai) [inline partitioned-table.md:1046]:
```suggestion
To convert the existing `member_level` table to a RANGE partitioned table, you can execute the following statement:
```

**Review by joey-yez — APPROVED** [review]

File: partitioned-table.md [hunk 8]
```diff
@@ -1084,6 +1084,46 @@ ALTER TABLE example TRUNCATE PARTITION p0;
...
+ALTER TABLE <table_name> REMOVE PARTITIONING
+```
+
+For example, to convert the `members` partitioned table to the non-partitioned table, you can execute the following statement:
+
+```sql
+ALTER TABLE members REMOVE PARTITIONING
+```
+
+## Partition an existing table
```
Comment (qiancai) [inline partitioned-table.md:1101]:
```suggestion
## Partition an existing table or modify partition type
```

**Review by hfxsd — APPROVED** [review]

**Review by qiancai — APPROVED** [review]

File: partitioned-table.md [hunk 9]
```diff
@@ -1084,6 +1084,46 @@ ALTER TABLE example TRUNCATE PARTITION p0;
 Query OK, 0 rows affected (0.03 sec)
 ```
 
+## Convert a partitioned table to a non-partitioned table
```
Comment (qiancai) [inline partitioned-table.md:1087]:
```suggestion
### Convert a partitioned table to a non-partitioned table
```

File: partitioned-table.md (hunk 8 above)
Comment (qiancai) [inline partitioned-table.md:1101]:
```suggestion
### Partition an existing table
```

**Pull request merged**